var/spack/repos/builtin/packages/eccodes/package.py

```diff
@@ -81,17 +81,15 @@ class Eccodes(CMakePackage):
     variant("shared", default=True, description="Build shared versions of the libraries")
 
     variant(
-        "definitions",
-        values=disjoint_sets(("auto",), ("default",) + tuple(_definitions.keys())).with_default(
-            "auto"
-        ),
-        description="List of definitions to install",
+        "extra_definitions",
+        values=any_combination_of(*_definitions.keys()),
+        description="List of extra definitions to install",
     )
 
     variant(
-        "samples",
-        values=disjoint_sets(("auto",), ("default",)).with_default("auto"),
-        description="List of samples to install",
+        "extra_samples",
+        values=any_combination_of(),
```

Comment:
Is this a typo?

Reply:
Let's say it is :)

Reply:
Sorry, I didn't get the reply. What do we want to achieve with a variant which can have any combination of `()`?

Reply:
I have already changed it to `any_combination_of(*_samples.keys())`. Although `_samples` is currently empty, it might be extended in the future.

Reply:
We already have variant `samples`, which should be either removed (because the default samples should always be installed) or renamed to `samples_extra` (similar to `definitions` -> `extra_definitions`). I decided to rename the variant and implement everything required to install extra samples (there is more coming with the follow-up PRs, see #29299 to get an idea of the final goal). So that if anybody wants to have extra samples, all they have to do is add an entry to `_samples`.

Reply:
> How do you see it used in practice?

The idea is that if somebody wants to add a set of extra samples, all they have to do is to come up with a simple PR that adds an entry to `_samples` (similar to #34208 but for samples).

However, I don't mind simply removing the `samples` variant in this PR. It will just be a bit annoying to recover the respective logic once/if it is needed.

Reply:
> However, I don't mind simply removing the samples variant in this PR. 

After that this is ready to be merged :+1: